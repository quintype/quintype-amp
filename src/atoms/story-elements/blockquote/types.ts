export interface blockquoteTypes {
  wrapperInlineStyles?: object;
  blockquoteInlineStyles?: object;
  attributionInlineStyles?: object;
  fallbackInlineStyles?: object;
}
