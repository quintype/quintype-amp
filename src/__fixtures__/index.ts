import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
