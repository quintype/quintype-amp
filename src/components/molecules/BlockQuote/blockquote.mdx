---
name: BlockQuote
menu: Story Elements
route: "/story-elements/blockquote"
---

### This is a blockquote