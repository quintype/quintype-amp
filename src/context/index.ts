// these are higher order components which wrap given component with story, config etc

import { withStory } from "./story/with-story";
import { withConfig } from "./config/with-config";
import { withStoryAndConfig } from "./story-and-config/with-story-config";

export { withStory, withConfig, withStoryAndConfig };
