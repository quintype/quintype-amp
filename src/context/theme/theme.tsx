import React from "react";
import { ThemeProvider } from "styled-components";
import defaultTokens from "./tokens";
import _merge from "lodash.merge";

import { ThemeProps } from "./theme-types";

const Theme = ({ tokens, children }: ThemeProps) => {
  const theme = tokens ? _merge({}, defaultTokens, tokens) : defaultTokens;

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { Theme };
