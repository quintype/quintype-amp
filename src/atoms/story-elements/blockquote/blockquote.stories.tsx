import * as React from "react";
import { storiesOf } from "@storybook/react";
import { BlockQuote } from "./blockquote";
import { config, textStory } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleBlockQuoteElement = {
  id: "1",
  type: "text",
  subtype: null,
  text: "<div><blockquote>This is a blockquote</blockquote><span class='attribution'>Attribution</span></div>",
  metadata: {
    content:
      "People talk about jails and prisons being incubators for infection. How does that work exactly on Rikers?",
    attribution: "Attribution"
  }
};

const { metadata, ...sampleBlockQuoteElementWithoutMetadata } = sampleBlockQuoteElement;

storiesOf("BlockQuote", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default blockquote", () => <BlockQuote element={sampleBlockQuoteElement} />)
  .add("Blockquote without metadata", () => <BlockQuote element={sampleBlockQuoteElementWithoutMetadata} />)
  .add("With custom styles ", () => (
    <BlockQuote
      element={sampleBlockQuoteElement}
      wrapperInlineStyles={{ backgroundColor: "ccf" }}
      blockquoteInlineStyles={{ color: "red" }}
      attributionInlineStyles={{ fontStyle: "italic" }}
      fallbackInlineStyles={{ color: "blue" }}
    />
  ));
