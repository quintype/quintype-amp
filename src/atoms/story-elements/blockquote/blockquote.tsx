import React from "react";
import styled from "styled-components";
import { StoryElementProps } from "../types";
import { Spacer } from "../../spacer";
import { blockquoteTypes } from "./types";
import { CommonRenderPropTypes } from "../../../types/config";
import { withStoryAndConfig } from "../../../context";

const Wrapper = styled.div.attrs(({ style }: StoryElementProps) => ({
  style
}))`
  display: flex;
  flex-direction: column;
  position: relative;

  &:before {
    content: "\\201C";
    font-size: ${(props) => props.theme.font.size.big};
    font-weight: bold;
    color: ${(props) => props.theme.color.mono7};
    margin: 0 ${(props) => props.theme.spacing.s};
    position: absolute;
    top: 4px;
  }
`;
const StyledBlockQuote = styled.blockquote.attrs(({ style }: StoryElementProps) => ({
  style
}))`
  padding: 0 0 0 50px;
  margin: 0;
  font-size: ${(props) => props.theme.font.size.l};
  color: ${(props) => props.theme.color.mono6};
  line-height: ${(props) => props.theme.font.lineHeight.level5};
`;
const StyledAttribution = styled.span.attrs(({ style }: StoryElementProps) => ({
  style
}))`
  color: ${(props) => props.theme.color.mono6};
  font-size: ${(props) => props.theme.font.size.m};
  font-weight: bold;
  padding: 0 0 0 38px;
  &:before {
    content: "-";
    margin-right: 5px;
  }
`;
const FallbackBlockQuote = styled.div.attrs(({ style }: StoryElementProps) => ({
  style
}))`
  div {
    display: flex;
    flex-direction: column;
    position: relative;

    &:before {
      content: "\\201C";
      font-size: ${(props) => props.theme.font.size.big};
      font-weight: bold;
      color: ${(props) => props.theme.color.mono7};
      margin: 0 ${(props) => props.theme.spacing.s};
      position: absolute;
      top: 4px;
    }
  }

  blockquote {
    padding: 0 0 0 50px;
    margin: 0;
    font-size: ${(props) => props.theme.font.size.l};
    color: ${(props) => props.theme.color.mono6};
    line-height: ${(props) => props.theme.font.lineHeight.level5};
  }

  .attribution {
    text-align: right;
    color: ${(props) => props.theme.color.mono6};
    font-size: ${(props) => props.theme.font.size.m};
    font-weight: bold;
    &:before {
      content: "-";
      margin-right: 5px;
    }
  }
`;

export const DefaultBlockQuote = ({
  element,
  wrapperInlineStyles,
  blockquoteInlineStyles,
  attributionInlineStyles,
  fallbackInlineStyles
}: StoryElementProps & blockquoteTypes) => {
  if (element.metadata) {
    const { content, attribution } = element.metadata;
    return (
      <Wrapper style={wrapperInlineStyles}>
        <StyledBlockQuote style={blockquoteInlineStyles}>{content}</StyledBlockQuote>
        {attribution && attribution.length && (
          <React.Fragment>
            <Spacer token="s" />
            <StyledAttribution style={attributionInlineStyles}>{attribution}</StyledAttribution>
          </React.Fragment>
        )}
      </Wrapper>
    );
  }

  return <FallbackBlockQuote style={fallbackInlineStyles} dangerouslySetInnerHTML={{ __html: element.text || "" }} />;
};

export const BlockQuoteBase = ({ element, story, config }: StoryElementProps & CommonRenderPropTypes) => {
  return config.opts && config.opts.blockquoteRender ? (
    config.opts.blockquoteRender({ story, config })
  ) : (
    <DefaultBlockQuote element={element} story={story} config={config} />
  );
};
export const BlockQuote = withStoryAndConfig(BlockQuoteBase);
