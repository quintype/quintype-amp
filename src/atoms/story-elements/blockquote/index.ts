import { BlockQuote } from "./blockquote";
export { BlockQuote };
