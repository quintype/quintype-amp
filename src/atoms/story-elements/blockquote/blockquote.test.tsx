import React from "react";
import { BlockQuoteBase, DefaultBlockQuote } from "./blockquote";
import { shallow, mount } from "enzyme";
import { textStory, config } from "../../../__fixtures__";
import { Theme } from "../../../context/theme";
const sampleBlockQuoteElement = {
  id: "1",
  type: "text",
  subtype: null,
  text: "<div><blockquote>This is a blockquote</blockquote><span class='attribution'>Attribution</span></div>",
  metadata: {
    content:
      "People talk about jails and prisons being incubators for infection. How does that work exactly on Rikers?",
    attribution: "Attribution"
  }
};

const { metadata, ...sampleBlockQuoteElementWithoutMetadata } = sampleBlockQuoteElement;

describe("BlockQuote", () => {
  it("should render default", () => {
    const wrapper = shallow(<DefaultBlockQuote element={sampleBlockQuoteElement} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render without metadata", () => {
    const wrapper = shallow(<DefaultBlockQuote element={sampleBlockQuoteElementWithoutMetadata} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call blockquoteRender prop when passed to opts", () => {
    const blockquoteRender = jest.fn();
    const modifiedConfig = {
      ...config,
      opts: { ...config.opts, storyElementRender: { blockquoteRender } }
    };
    const wrapper = shallow(
      <BlockQuoteBase element={sampleBlockQuoteElement} story={textStory} config={modifiedConfig} />
    );
    expect(blockquoteRender.mock.calls.length).toBe(1);
    expect(wrapper.find(DefaultBlockQuote).length).toBe(0);
  });
});
