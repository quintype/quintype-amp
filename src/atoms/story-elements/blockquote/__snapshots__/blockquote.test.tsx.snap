// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockQuote should render default 1`] = `
<styled.div
  textDirection="ltr"
>
  <styled.blockquote
    textDirection="ltr"
  >
    People talk about jails and prisons being incubators for infection. How does that work exactly on Rikers?
  </styled.blockquote>
  <Spacer
    token="s"
  />
  <styled.span>
    Attribution
  </styled.span>
</styled.div>
`;
